R/disease_progression.R

```diff
@@ -6,11 +6,22 @@
 #' @param recovery_outcome competing hazards object for recovery rates
 #' @noRd
 create_recovery_rates_process <- function(
+  parameters,
   variables,
   recovery_outcome
 ) {
   function(timestep){
-    recovery_outcome$set_rates(variables$recovery_rates$get_values())
+    recovery_rates <- variables$recovery_rates$get_values()
+    if (parameters$parasite == "vivax"){
+      # p.v subpatent recovery is immunity-dependent
+      recovery_rates[variables$state$get_index_of("U")$to_vector()] <-
+        1/anti_parasite_immunity(
+          parameters$dpcr_min, parameters$dpcr_max, parameters$apcr50, parameters$kpcr,
+          variables$iaa$get_values(index = variables$state$get_index_of("U")),
+          variables$iam$get_values(index = variables$state$get_index_of("U"))
+        )
```

Comment:
Hard to parse like this, I think we can make it more efficient too (untested suggestion).
```suggestion
  # p.v subpatent recovery is immunity-dependent
  
  U_index <- variables$state$get_index_of("U")
  
  recovery_rates[U_index$to_vector()] <-
    1 / anti_parasite_immunity(
      min = parameters$dpcr_min,
      max = parameters$dpcr_max,
      a50 = parameters$apcr50,,
      k = parameters$kpcr,
      iaa = variables$iaa$get_values(index = U_index),
      iam =variables$iam$get_values(index = U_index)
    )
```